IdentityCore/src/util/NSError+MSIDThrottlingExtension.m

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import "NSError+MSIDThrottlingExtension.h"
+#import "NSError+MSIDExtensions.h"
+
+@implementation NSError (MSIDThrottlingExtension)
+
+- (NSDate *)msidGetRetryDateFromError
+{
+    if (!self) return nil;
+    NSDate *retryHeaderDate = nil;
+    NSString *retryHeaderString = nil;
+    NSDictionary *headerFields = [self.domain hasPrefix:@"MSID"] ? self.userInfo[MSIDHTTPHeadersKey] : self.userInfo[@"MSALHTTPHeadersKey"];
```

Comment:
I see the check for [self.domain hasPrefix:@"MSID"] in multiple places. Consider having a utility that does the mapping/error operation for you. 

e.g. [MSIDErrorUtil getHTTPHeadersKeyForError:] - and it returns you correct key or value depending on the type

Reply:
Good suggestion, update in next commit.